### PR TITLE
This patches fixes following;

### PR DESCRIPTION
**Pull request by adeelkhan**
*) Empty fields error messages
*) Removes onChange validations

VAN-204

### CODE REVIEW

File: src/logistration/RegistrationPage.jsx [hunk 1]
```diff
@@ -192,24 +215,44 @@ class RegistrationPage extends React.Component {
 
     switch (inputName) {
       case 'email':
-        emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
-        errors.email = emailValid ? '' : this.intl.formatMessage(messages['logistration.email.validation.message']);
+        emailValid = value.length >= 1;
+        if (!emailValid) {
+          emptyFields.email = [{ user_message: this.intl.formatMessage(messages['logistration.email.validation.message']) }];
+        } else {
+          emptyFields.email = [{ user_message: '' }];
```
Comment (waheedahmed) [inline src/logistration/RegistrationPage.jsx:222]: Can we handle all these errors using a utility function like the following?
```
  generateUserMessage = (isFieldValid, messageID) => {
    return [{ user_message: isFieldValid ? '' : this.intl.formatMessage(messages[messageID] }];
  }
```
```suggestion
        emptyFields.email = generateUserMessage(emailValid, 'logistration.email.validation.message']);
```

File: src/logistration/RegistrationPage.jsx [hunk 2]
```diff
@@ -96,8 +97,31 @@ class RegistrationPage extends React.Component {
       const { errors } = this.state;
       const errorMsg = nextProps.validations.validation_decisions[this.state.validationFieldName];
       errors[this.state.validationFieldName] = errorMsg;
-      this.setState({
-        errors,
+      const stateValidVar = `${camelCase(this.state.validationFieldName)}Valid`;
```
Comment (waheedahmed) [inline src/logistration/RegistrationPage.jsx:100]: nit: `stateValidKey`

File: src/logistration/RegistrationPage.jsx [hunk 3]
```diff
@@ -96,8 +97,31 @@ class RegistrationPage extends React.Component {
...
+      const stateValidValue = !errorMsg;
+
+      this.setState(() => ({ [stateValidVar]: stateValidValue }), () => {
```
Comment (waheedahmed) [inline src/logistration/RegistrationPage.jsx:103]: nit:
```suggestion
      this.setState(({ [stateValidVar]: stateValidValue }), () => {
```

**Review by waheedahmed — CHANGES_REQUESTED** [review]
Also, React suggests that we should avoid extensive operations inside `shouldComponentUpdate()`, which can harm the performance. Is there any other way of doing what we are currently doing inside it?

File: src/logistration/RegistrationPage.jsx (hunk 3 above)
Comment (adeelkhan) [reply src/logistration/RegistrationPage.jsx]: Done

File: src/logistration/RegistrationPage.jsx (hunk 2 above)
Comment (adeelkhan) [reply src/logistration/RegistrationPage.jsx]: Done

File: src/logistration/RegistrationPage.jsx (hunk 1 above)
Comment (adeelkhan) [reply src/logistration/RegistrationPage.jsx]: Done

File: src/logistration/RegistrationPage.jsx [hunk 4]
```diff
@@ -192,24 +215,44 @@ class RegistrationPage extends React.Component {
...
+        if (!usernameValid) {
+          emptyFields.username = [{ user_message: this.intl.formatMessage(messages['logistration.username.validation.message']) }];
+        } else {
+          emptyFields.username = [{ user_message: '' }];
+        }
         break;
       case 'password':
-        passwordValid = !!(value.length >= 8 && value.match(/\d+/g));
-        errors.password = passwordValid ? '' : this.intl.formatMessage(messages['logistration.register.page.password.validation.message']);
+        passwordValid = !!(value.length >= 1);
```
[the remark this answers is not in the document]
Comment (adeelkhan) [reply src/logistration/RegistrationPage.jsx]: Done

**Review by adeelkhan — COMMENTED** [review]
> Also, React suggests that we should avoid extensive operations inside `shouldComponentUpdate()`, which can harm the performance. Is there any other way of doing what we are currently doing inside it?

Two things here; 
1) The working here are not too expensive in nature. Just setting state and in async testing form validation.
2) I have to update state before render, and this is one of way. I don't know for any other solution for now. We can fix forward.

File: src/logistration/_style.scss [hunk 5]
```diff
@@ -218,3 +218,7 @@ $apple-focus-black: $apple-black;
 .opt-year-field{
     padding-left: 15px;
 }
+
+.invalid-feedback {
+  color: #b20610
```
Comment (waheedahmed) [inline src/logistration/_style.scss:223]: Use theme variables `$danger` or `$red`.

**Review by waheedahmed — APPROVED** [review]
One minor change otherwise 👍

File: src/logistration/_style.scss (hunk 5 above)
Comment (adeelkhan) [reply src/logistration/_style.scss]: Done

**Pull request merged**